pwiz_tools/Shared/Common/Collections/NaturalComparer.cs

```diff
@@ -0,0 +1,102 @@
+﻿/*
+ * Original author: Clark Brace <clarkbrace@gmail.com>,
+ *                  MacCoss Lab, Department of Genome Sciences, UW
+ *
+ * Copyright 2022 University of Washington - Seattle, WA
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace pwiz.Common.Collections
+{
+    /// <summary>
+    /// Added by Clark Brace (cbrace3)
+    /// Code sourced form Justin.Jones as part of (CPOL)
+    /// https://www.codeproject.com/Articles/22517/Natural-Sort-Comparer
+    /// </summary>
+    public class NaturalComparer
+    {
+
+        public static int Compare(string x, string y)
+        {
+            if (x == null && y == null)
+            {
+                return 0;
+            }
+
+            if (x == null)
+            {
+                return 1;
+            }
+
+            if (y == null)
+            {
+                return -1;
+            }
+
+
+            var x1 = Regex.Split(x, @"([0-9]+)");
+
+            var y1 = Regex.Split(y, @"([0-9]+)");
+
+
+            for (var i = 0; i < x1.Length && i < y1.Length; i++)
+            {
+                var left = x1[i];
+                var right = y1[i];
+                // If both values are numbers
+                if (int.TryParse(left, out var l) && int.TryParse(right, out var r))
+                {
+                    var comp  =l.CompareTo(r);
+                    if (comp != 0)
+                    {
+                        return comp;
+                    }
+                }
+                else if (!int.TryParse(left, out _) || !int.TryParse(right, out _)) // If both are letters
```

Comment:
Let's try cacheing these TryParse results - that's a fairly expensive call, this could start to be a performance issue in a long list

Reply:
Actually Nick is of the opinion that we should just replace all this with a call to the Windows implementation, that's probably not a bad idea, and an interesting thing to learn how to do. Look for "DLLImport" for examples of that in our code.

Reply:
https://www.pinvoke.net/default.aspx/shlwapi.strcmplogicalw